aea/cli/search.py

```diff
@@ -62,47 +62,52 @@ def search(click_context, local):
 
 @search.command()
 @click.option("--query", default="", help="Query string to search Connections by name.")
+@click.option("--page", type=int, default=1, help="Page number to display.")
 @pass_ctx
-def connections(ctx: Context, query):
+def connections(ctx: Context, query, page):
     """Search for Connections."""
     item_type = "connection"
-    _output_search_results(item_type, search_items(ctx, item_type, query))
+    _output_search_results(item_type, *search_items(ctx, item_type, query, page), page)
 
 
 @search.command()
 @click.option("--query", default="", help="Query string to search Contracts by name.")
+@click.option("--page", type=int, default=1, help="Page number to display.")
 @pass_ctx
-def contracts(ctx: Context, query):
+def contracts(ctx: Context, query, page):
     """Search for Contracts."""
     item_type = "contract"
-    _output_search_results(item_type, search_items(ctx, item_type, query))
+    _output_search_results(item_type, *search_items(ctx, item_type, query, page), page)
 
 
 @search.command()
 @click.option("--query", default="", help="Query string to search Protocols by name.")
+@click.option("--page", type=int, default=1, help="Page number to display.")
 @pass_ctx
-def protocols(ctx: Context, query):
+def protocols(ctx: Context, query, page):
     """Search for Protocols."""
     item_type = "protocol"
-    _output_search_results(item_type, search_items(ctx, item_type, query))
+    _output_search_results(item_type, *search_items(ctx, item_type, query, page), page)
 
 
 @search.command()
 @click.option("--query", default="", help="Query string to search Skills by name.")
+@click.option("--page", type=int, default=1, help="Page number to display.")
 @pass_ctx
-def skills(ctx: Context, query):
+def skills(ctx: Context, query, page):
     """Search for Skills."""
     item_type = "skill"
-    _output_search_results(item_type, search_items(ctx, item_type, query))
+    _output_search_results(item_type, *search_items(ctx, item_type, query, page), page)
 
 
 @search.command()
 @click.option("--query", default="", help="Query string to search Agents by name.")
+@click.option("--page", type=int, default=1, help="Page number to display.")
```

Comment:
you could consider creating a composed annotator to avoid code duplication.

Reply:
@marcofavorito how can I do that?
And actually that's not a part of this PR as we have already `--query` described for all commands separately. Would be nice if you will create a ticket and I will do optimization for those commands in other PR.

Reply:
basically, just define a function with all the decorators in one place (I'd say at the top of the module). But you're right, it's out of the scope for this PR.